lightly/loss/byol_loss.py

```diff
@@ -0,0 +1,189 @@
+import singleton as singleton
+import torch
+from torch import nn
+
+import torch.nn.functional as F
+
+from models.mlp import MLP
+
+
+class ExperimentParams:
+    """
+    Encapsulate hyper-parameters for performing and logging experiments.
+    """
+
+    def __init(self, use_momentum, beta, augment_functions, net, projection_size, projection_hidden_size, layer):
+        # TODO: implement momentum
+        self.use_momentum = use_momentum
+        self.beta = beta
+        self.augment_functions = augment_functions
+        self.net = net
+        self.projection_size = projection_size
+        self.projection_hidden_size = projection_hidden_size
+        self.layer = layer
+
+    def __str__(self):
+        return "Running experiment with the following hyper-parameters.\n" \
+               "Momentum: {}\nBeta: {}\nAugment Function: {}\nNet: {}\nProjection Size: {}\n" \
+               "Projection Hidden Size: {}\n" "Layer: {}".format(self.use_momentum, self.beta, self.augment_functions,
+                                                                 self.net, self.projection_size,
+                                                                 self.projection_hidden_size, self.layer)
+
+
```

Comment:
This is a cool idea! We still have to find the best way to get the activations from intermediate levels of the backbone net. However, I don't think `byol_loss.py` is the right place for such a feature.

Please remove `NetWrapper` from this file and feel free to make a suggestion in #60 on how to solve this issue.